controllers/metallb_controller.go

```diff
@@ -95,16 +95,17 @@ func (r *MetallbReconciler) syncMetalLBResources(config *metallbv1alpha1.Metallb
 		return err
 	}
 
+	var applyErr error
 	for _, obj := range objs {
 		if err := controllerutil.SetControllerReference(config, obj, r.Scheme); err != nil {
 			return errors.Wrapf(err, "Failed to set controller reference to %s %s", obj.GetNamespace(), obj.GetName())
 		}
 
 		// Open question: should an error here indicate we will never retry?
 		if err := apply.ApplyObject(context.TODO(), r.Client, obj); err != nil {
-			err = errors.Wrapf(err, "could not apply (%s) %s/%s", obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
+			applyErr = errors.Wrapf(err, "could not apply (%s) %s/%s", obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
+			logger.Error(applyErr, "Failed to apply object")
```

Comment:
Just return here, no need to continue with the loop

Reply:
why logging error here and not at the caller if err != nil ?

Reply:
Good point. Also, I believe that if Reconcile returns an error, the framework will log that.
@mmirecki can you try to force this to return the error with no logs and see if that is dumped? If so, we can just let the error float

Reply:
We'll now exit the loop on first failure.
All wrapped errors are dumped.